packages/components/focus-trap/src/focus-trap.vue

```diff
@@ -236,14 +276,19 @@ export default defineComponent({
       if (trapContainer) {
         trapContainer.removeEventListener(FOCUS_AFTER_TRAPPED, trapOnFocus)
 
-        const releasedEvent = new Event(
-          FOCUS_AFTER_RELEASED,
-          FOCUS_AFTER_TRAPPED_OPTS
-        )
+        const releasedEvent = new CustomEvent(FOCUS_AFTER_RELEASED, {
+          ...FOCUS_AFTER_TRAPPED_OPTS,
+          detail: {
+            focusReason: focusReason.value,
+          },
+        })
         trapContainer.addEventListener(FOCUS_AFTER_RELEASED, releaseOnFocus)
         trapContainer.dispatchEvent(releasedEvent)
 
-        if (!releasedEvent.defaultPrevented) {
+        if (
+          !releasedEvent.defaultPrevented &&
+          (focusReason.value == 'keyboard' || !isFocusCausedByUserEvent())
```

Comment:
我有个建议，把这里的290行改成`trapContainer.contains(document.activeElement)`

Reply:
It's not this simple, because many components purposely trigger exiting the trap by putting the focus outside of the container. Focus trap's job is to recognize that happened and correct the focus.